craft_parts/plugins/spring_boot_plugin.py

```diff
@@ -0,0 +1,196 @@
+        """Return a dictionary with the environment to use in the build step."""
+        return {}
+
+    @override
+    def get_build_snaps(self) -> set[str]:
+        """Return a set of required snaps to install in the build environment."""
+        return set()
+
+    @override
+    def get_build_commands(self) -> list[str]:
+        """Return a list of commands to run during the build step."""
+        commands = []
+        # Set JAVA_HOME
+        commands.append(
+            textwrap.dedent(
+                """\
+                if [ -z "${JAVA_HOME+x}" ]; then
+                    JAVA_HOME=$(dirname $(dirname $(readlink -f $(which java))))
+                fi
+                """
+            )
+        )
+        # Set project base
+        commands.append(
+            textwrap.dedent(
+                """\
+                if [[ -e "./mvnw" ]]; then
+                    PROJECT_BASE="maven"
+                elif [[ -e "./gradlew" ]]; then
+                    PROJECT_BASE="gradle"
+                else
+                    echo 'Neither "mvnw" nor "gradlew" found.'
+                    exit 1
+                fi
+                """
+            )
+        )
+        # Check executability of wrapper files
+        commands.append(
+            textwrap.dedent(
+                """\
+                if [[ "$PROJECT_BASE" == "maven" && ! -x "./mvnw" ]]; then
+                    echo '"mvnw" found but not executable.'
+                    exit 1
+                elif [[ "$PROJECT_BASE" == "gradle" && ! -x "./gradlew" ]]; then
+                    echo '"gradlew" found but not executable.'
+                    exit 1
+                fi
+                """
+            )
+        )
+
+        # Check system Java version compatibility with project Java version
+        commands.append(
+            textwrap.dedent(
+                """\
+                SYSTEM_JAVA_MAJOR_VERSION=$($JAVA_HOME/bin/java -version 2>&1 | grep -oP \
+                'openjdk version "\\K(1\\.\\d+|\\d+)')
+                """
+            )
+        )
+        commands.append(
+            textwrap.dedent(
+                """\
+                if [[ "$PROJECT_BASE" == "maven" ]]; then
+                    PROJECT_JAVA_VERSION=$("./mvnw" help:evaluate -Dexpression=project.properties \
+                -q -DforceStdout | grep -oP '<java\\.version>\\K(.*?)(?=</java\\.version>)')
+                elif [[ "$PROJECT_BASE" == "gradle" ]]; then
+                    PROJECT_JAVA_VERSION=$("./gradlew" javaToolchains | grep -oP \
+                    "Language Version:\\s+\\K(\\d+)")
+                fi
+                """
+            )
+        )
+        commands.append(
+            textwrap.dedent(
+                """\
+                if [[ $PROJECT_JAVA_VERSION -gt $SYSTEM_JAVA_MAJOR_VERSION ]]; then
+                    echo "Project requires Java version $PROJECT_JAVA_VERSION, but system Java \
+                version is $SYSTEM_JAVA_MAJOR_VERSION."
```

Comment:
Let's add a link explaining that the project requires a Java version which is later than is available in the system which means the app won't start